lisa/sut_orchestrator/azure/common.py

```diff
@@ -576,7 +973,7 @@ def get_image_name(self) -> str:
             assert isinstance(
                 self.shared_gallery_raw, dict
             ), f"actual type: {type(self.shared_gallery_raw)}"
-            if self.shared_gallery.resource_group_name:
+            if self.shared_gallery_raw.get("resource_group_name", ""):
```

Comment:
Why get it from raw?

Reply:
I could change `shared_gallery_raw.get` to `shared__gallery.__dict__.get` if you want both of them to pull from the object and now the raw.

Reply:
I cannot figure out what happened from the code reviewing. Can you share why the raw and object unsync? We should fix the unsync. The raw info is used in only one scenario, but I forgot when it's used.

Reply:
I fixed the syncing, but how do you want me to move forward with this logic. My original change was because the check `if self.shared_gallery.resource_group_name` does not match the variable being used `[self.shared_gallery_raw.get(k, "") for k in SIG_IMAGE_KEYS]`

I still think both should use the same, either `shared_gallery` or `shared_gallery_raw`

Reply:
Ideally, they should be consistent, so `self.shared_gallery.resource_group_name` is better, because it's a var name, not a string.
Let me know, if in some cases it's inconsistent. It needs to understand why not consistent and fix them.